notebooks/customizing_potentials_cookbook.ipynb

```diff
@@ -0,0 +1,1780 @@
+        "\n",
+        "As an important example, here we consider the case where the energy includes both a pair potential and a bond potential. Specifically, we combine ```harmonic_morse_pair``` with ```bistable_spring_bond```. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "KowW2fMuV_nY",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "#Note, the code in the \"Bonds\" section must be run prior to this.\n",
+        "energy_fn = harmonic_morse_pair(displacement,D0=0.,alpha=10.0,r0=1.0,k=1.0)\n",
+        "bond_energy_fn = bistable_spring_bond(displacement, bonds, r0=lengths)\n",
+        "def combined_energy_fn(R):\n",
+        "  return energy_fn(R) + bond_energy_fn(R)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LR8NW_NkZV0y",
+        "colab_type": "text"
+      },
+      "source": [
+        "Here, we have set $D_0=0$, so the pair potential is just a one-sided repulsive harmonic potential. For particles connected with a bond, this raises the energy of the \"contracted\" minimum relative to the \"extended\" minimum."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "vNd0A8CYXCuR",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "drs = np.arange(0,2,0.01)\n",
+        "U = harmonic_morse(drs,D0=0.,alpha=10.0,r0=1.0,k=1.0)+bistable_spring(drs)\n",
+        "#F = -vmap(grad(combined_energy_fn))(drs)\n",
+        "plt.plot(drs,U)\n",
+        "#plt.plot(drs,F)\n",
+        "format_plot(r'$r$', r'$V(r)$')\n",
+        "finalize_plot()"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "fMU9ethYZzry",
+        "colab_type": "text"
+      },
+      "source": [
+        "This new energy function can be passed to the minimization routine (or any other JAX MD simulation routine) in the usual way."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "h67C8SRrWm8J",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "Rfinal, max_force_component = run_minimization(combined_energy_fn, R_temp, shift)\n",
+        "print('largest component of force after minimization = {}'.format(max_force_component))\n",
+        "plot_system( Rfinal, box_size )"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "7JkkgEY4H19L",
+        "colab_type": "text"
+      },
+      "source": [
+        "#Specifying forces instead of energies"
```

Comment:
Pinning!!!

Reply:
Yes! :) Though the better way would be to use a custom shift function... Should I explicitly show that?